### ISSUE TITLE
get-aduser doc:  -filter is type string, any scriptblock references are incorrect

### ISSUE THREAD

**Issue by jszabo98** [MicrosoftDocs/windows-powershell-docs#1584]
Since the -filter option is documented as type "string", any references to scriptblocks are incorrect, such as example 4, and the Backus-Naur form under "-Filter".

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 39c80552-2be1-2869-5ea8-2368bb3dd2c8
* Version Independent ID: 8d825bb9-93d8-16ca-22f0-a8191d642a50
* Content: [Get-ADUser (addsadministration)](https://docs.microsoft.com/en-us/powershell/module/addsadministration/get-aduser?view=win10-ps)
* Content Source: [docset/windows/addsadministration/Get-ADUser.md](https://github.com/MicrosoftDocs/windows-powershell-docs/blob/live/docset/windows/addsadministration/Get-ADUser.md)
* Product: **w10**
* Technology: **powershell-windows**
* GitHub Login: @andreabarr
* Microsoft Alias: **v-anbarr**

**Comment by velkovb** [comment]
@jszabo98 Hello

Thank you for trying to make MicrosoftDocs a better place.

The parameter is defined as string indeed but it accepts the described syntax in the "-Filter" section. It is not a script block but a string parsed according to these rules. Here is a simple example:
![image](https://user-images.githubusercontent.com/10958657/81501606-fd1c2b00-92e1-11ea-8a5d-d87a7a4e348a.png)

Let me know if you have any questions.

**Comment by e0i** [comment]
@jszabo98 

The issue is being closed since your question seems to be answered. Feel free to re-open this issue if you feel that it hasn't been answered or that there are further suggestions to improve the documentation.

Thanks.

**Comment by jszabo98** [comment]
It's a scriptblock that's coerced into a string.  I think it's strange that the official docs have an example like that.  

```powershell
$sb = { name -eq 'test' }
$sb.gettype()

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     ScriptBlock                              System.Object
```

There are some lesser known examples that won't work with a scriptblock.

```powershell
$obj = [pscustomobject]@{name = 'test'}

get-aduser { name -eq $($obj.name) }  # doesn't work

get-aduser " name -eq '$($obj.name)' }  # works
```

**Comment by velkovb** [comment]
@jszabo98 
That is because you are using _pscustomobject_. Try the following:
![image](https://user-images.githubusercontent.com/10958657/82118721-cb8fdd80-9781-11ea-8136-dcf986eda70d.png)

You are correct that the -Filter parameter is of type filter and everything passed to it will be casted to string. So if you want to see how the script block will be passed to the command you can do the following:
````powershell
[string]{ name -eq $user.name }
````

The reason it is not working with _pscustomobject_ is that _name_ is a _noteproperty_ while in the ADUser object it is a regular _property_. It think if you want to do that you need to dig into classes :)
![image](https://user-images.githubusercontent.com/10958657/82118900-c8492180-9782-11ea-895a-e0e9a5c9150e.png)

Let me know if this was helpful.

**Comment by jszabo98** [comment]
That's odd that the aduser object works, but not the pscustomobject.  I wish I understood what was going on with ad filters.  They seem like a black box.  However, when I try the same thing, I get no result.  Also, what if someone had an object from importing a csv, for example?  

```powershell
$user = get-aduser -filter { name -eq 'jszabo' }
get-aduser -filter { user -eq $user.name }  # no result

get-command get-aduser

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Cmdlet          Get-ADUser                                         1.0.1.0    ActiveDirectory
```

**Comment by velkovb** [comment]
Interesting, what is your PSVersion and OSVersion? I am testing on Server 2016 core.

**Comment by jszabo98** [comment]
You're right, it's working with "name -eq".  The string versions of either scriptblock are exactly the same.  The code must be going back to the variables it its scope using invoke-expression or something to find what the original $user object is.  Do you know where the source code is?

**Comment by velkovb** [comment]
As far as I know the AD module is not opensource. The difference is in the objects, no magic here. Type Get-Member for each of them and you will see the pscustomobject has noteproperty and the aduser object has property. Noteproperties do not seem to work.

**Comment by jszabo98** [comment]
It's just unusual that the filter is passed in as a string, not an object.

**Comment by velkovb** [comment]
The _Identity_ parameter accepts object as input but probably it will not work with the pscustomobject as well.

**Comment by officedocspr5** [comment]
To make it easier for you to submit feedback on articles on learn.microsoft.com, we're transitioning our feedback system from GitHub Issues to a new experience. 

As part of the transition, this GitHub Issue will be moved to a private repository. We're moving Issues to another repository so we can continue working on Issues that were open at the time of the transition. When this Issue is moved, you'll no longer be able to access it.

If you want to provide additional information before this Issue is moved, please update this Issue before December 15th, 2023.

With the new experience, you no longer need to sign in to GitHub to enter and submit your feedback. Instead, you can choose directly on each article's page whether the article was helpful. Then you can then choose one or more reasons for your feedback and optionally provide additional context before you select **Submit**.

Here's what the new experience looks like.

**Note**: The new experience is being rolled out across learn.microsoft.com in phases. If you don't see the new experience on an article, please check back later.

First, select whether the article was helpful:

![Image showing a dialog asking if the article was helpful with yes and no answers.](https://learn.microsoft.com/Office/media/internal/new-feedback-yes-no.jpg)

Then, choose at least one reason for your feedback and optionally provide additional details about your feedback:

| Article was helpful | Article was unhelpful |
|:--|:---|
|![Image showing a dialog asking how the article was helpful with several options.](https://learn.microsoft.com/Office/media/internal/new-feedback-helpful.jpg) | ![Image showing a dialog asking how the article wasn't helpful with several options.](https://learn.microsoft.com/Office/media/internal/new-feedback-unhelpful.jpg)|

Finally, select **Submit** and you're done!
